### DISCUSSION TITLE
unable to install cx_Oracle on AIX 7.2

### DISCUSSION THREAD

**Discussion by myOfficialWorkSpace** [oracle/python-oracledb#396] [Q&A]
Unable to install cx_Oracle on AIX 7.2. Please note that the server does not have any internet connection and pip command does not work there.

We did try to install using the below options -
python3.8 -m pip install --no-build-isolation cx_Oracle-8.3.0.tar.gz.

But getting the below error -
```
root@...........:/tmp/python > /opt/bin/python3.8 -m pip install --no-build-isolation cx_Oracle-8.3.0.tar.gz
Processing ./cx_Oracle-8.3.0.tar.gz
Preparing wheel metadata ... error
ERROR: Command errored out with exit status 1:
command: /opt/bin/python3.8 /opt/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpmlgqr7u6
cwd: /tmp/pip-req-build-mhyc7_ej
Complete output (13 lines):
running dist_info
creating /tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info
writing /tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info/PKG-INFO
writing dependency_links to /tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info/dependency_links.txt
writing top-level names to /tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info/top_level.txt
writing manifest file '/tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info/SOURCES.txt'
reading manifest file '/tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'odpi/test'
no previously-included directories found matching 'odpi/samples'
writing manifest file '/tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.egg-info/SOURCES.txt'
creating '/tmp/pip-req-build-mhyc7_ej/pip-wheel-metadata/cx_Oracle.dist-info'
error: invalid command 'bdist_wheel'
----------------------------------------
ERROR: Command errored out with exit status 1: /opt/bin/python3.8 /opt/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_
```

Can anyone help with this please?

**Comment by cjbj** [comment]
To be passive aggressive, and try and nudge you to upgrade, you should be installing python-oracledb, since it is cx_Oracle under a new name and with better features. See the [python-oracledb release announcement](https://cjones-oracle.medium.com/open-source-python-thin-driver-for-oracle-database-e82aac7ecf5a) from _over two years ago_.  If your AIX version is one of the platforms where the Python cryptography package won't build easily, then see [Installing python-oracledb without the Cryptography Package](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-python-oracledb-without-the-cryptography-package).

Regarding the error, there are various Google hits for it, e.g. [this](https://stackoverflow.com/a/44862371/4799035). 

Finally this current repo is for python-oracledb - the cx_Oracle repo is [here](https://github.com/oracle/python-cx_Oracle/issues).
